examples/custom_object3.py

```diff
@@ -48,16 +48,71 @@ def color(self, color):
         self.uniform_buffer.update_range(0, 99999)
 
 
+@gfx.renderers.wgpu.register_wgpu_render_function(Triangle, TriangleMaterial)
 class TriangleShader(WorldObjectShader):
+
+    type = "render"
+
+    def get_resources(self, wobject, shared):
+        geometry = wobject.geometry
+
+        # Set templating variable "scale" to 0.2
+        self["scale"] = 0.2
+
+        # Three uniforms and one storage buffer with positions
+        bindings = {
+            0: Binding("u_stdinfo", "buffer/uniform", shared.uniform_buffer),
+            1: Binding("u_wobject", "buffer/uniform", wobject.uniform_buffer),
+            2: Binding("u_material", "buffer/uniform", wobject.material.uniform_buffer),
+            3: Binding(
+                "s_positions", "buffer/read_only_storage", geometry.positions, "VERTEX"
+            ),
+        }
+        self.define_bindings(0, bindings)
+        return {
+            "index_buffer": None,
+            "vertex_buffers": {},
+            "bindings": {
+                0: bindings,
+            },
+        }
+
+    def get_pipeline_info(self, wobject, shared):
+        # We draw triangles, no culling
+        return {
+            "primitive_topology": wgpu.PrimitiveTopology.triangle_list,
+            "cull_mode": wgpu.CullMode.none,
+        }
+
+    def get_render_info(self, wobject, shared):
+        material = wobject.material
+        geometry = wobject.geometry
+
+        # Determine number of vertices
+        n = 3 * geometry.positions.nitems
+
+        # Define in what passes this object is drawn.
+        render_mask = wobject.render_mask
+        if not render_mask:  # i.e. set to auto
+            if material.is_transparent:
+                render_mask = RenderMask.transparent
+            else:
+                render_mask = RenderMask.opaque
```

Comment:
I think you should add a subsection to `docs/shaders.rst` to explain which render passes exist and how the rendermask relates to them.

Reply:
Good point, done!